lab_1/main.py

```diff
@@ -62,7 +97,17 @@ def compare_profiles(unknown_profile: dict, profile_to_compare: dict, top_n: int
     :param top_n: a number of the most common words
     :return: the distance
     """
-    pass
+    if type(unknown_profile) == dict and type(profile_to_compare) == dict and type(top_n) == int:
+        top_of_unknown_profile = get_top_n_words(unknown_profile['frequency'], top_n)
+        top_of_profile_to_compare = get_top_n_words(profile_to_compare['frequency'], top_n)
+        shared_words = []
+        for word in top_of_unknown_profile:
+            if word in top_of_profile_to_compare:
+                shared_words.append(word)
+                distance = len(shared_words) / len(top_of_unknown_profile)
```

Comment:
Are you sure about calculating the distance on each iteration of the `for word in top_of_unknown_profile:` loop?